From da86f5121a4024ea2a45eeaeb550f6f554928c8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Vitor=20de=20Lima=20Matos?=
 <andre.vmatos@gmail.com>
Date: Wed, 1 Nov 2023 11:58:54 -0300
Subject: [PATCH] kernel: fix framebuffer console after 6.6

These options were already enabled in previous versions, but Kconfig
changes in 6.6 made NixOS kernel disable them. Therefore, we enable
unconditionally, to be explicit that they're needed.
Without them, the fbcon/console on systems with DRM devices freeze/blank
on early boot, even though the system boots normally.
---
 pkgs/os-specific/linux/kernel/common-config.nix | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pkgs/os-specific/linux/kernel/common-config.nix b/pkgs/os-specific/linux/kernel/common-config.nix
index 0f59d3ac7aad7f..4e366a87d09e4f 100644
--- a/pkgs/os-specific/linux/kernel/common-config.nix
+++ b/pkgs/os-specific/linux/kernel/common-config.nix
@@ -320,9 +320,11 @@ let
       FRAMEBUFFER_CONSOLE = yes;
       FRAMEBUFFER_CONSOLE_DEFERRED_TAKEOVER = yes;
       FRAMEBUFFER_CONSOLE_ROTATION = yes;
+      FRAMEBUFFER_CONSOLE_DETECT_PRIMARY = yes;
       FB_GEODE            = mkIf (stdenv.hostPlatform.system == "i686-linux") yes;
       # On 5.14 this conflicts with FB_SIMPLE.
       DRM_SIMPLEDRM = whenAtLeast "5.14" no;
+      DRM_FBDEV_EMULATION = yes;
     };
 
     fonts = {
